### ISSUE TITLE
Is it safe to ignore this warning, or should I add sync_dist=True?

### ISSUE THREAD

**Issue by FT-Gyeongsu** [roboflow/rf-detr#909]
1. When training and validating with multiple GPUs, is sync_dist set to False by default?
2. Is there a specific place or configuration where it can be set?
3. Should I add sync_dist=True to that code?”

**Warning**
```python
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_ce', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/class_error', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_bbox', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_giou', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/cardinality_error', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_ce_0', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_bbox_0', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_giou_0', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/cardinality_error_0', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_ce_1', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_bbox_1', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_giou_1', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/cardinality_error_1', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_ce_enc', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_bbox_enc', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss_giou_enc', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/cardinality_error_enc', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
C:\Users\software\anaconda3\envs\rf_detr\lib\site-packages\pytorch_lightning\trainer\connectors\logger_connector\result.py:433: It is recommended to use `self.log('train/loss', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices.
```

**Comment by Borda** [comment]
Hi @FT-Gyeongsu — the warning is safe to ignore. It's informational, not an error — your training is working correctly.

What's happening: RF-DETR defaults to `sync_dist=False` for training loss metrics to avoid the communication overhead of an `all_reduce` on every log call. In multi-GPU mode, PyTorch Lightning flags this because epoch-level metrics won't be averaged across GPUs. But for training loss monitoring, each rank's local loss is a valid signal — your gradients, model weights, and validation metrics (which already use `sync_dist=True`) are unaffected.

If you'd like globally-averaged training metrics and want to silence the warning, there's a built-in config knob:

```python
model.train(
    dataset_dir="...",
    train_log_sync_dist=True,
)
```

This adds a small `all_reduce` per logging call — negligible for most setups.
